WeatherSystem.Test/WeatherBotShould.cs

```diff
@@ -0,0 +1,182 @@
+﻿using AutoFixture;
+using WeatherSystem.Factory;
+using WeatherSystem.Models;
+
+namespace WeatherSystem.Test
+{
+    public class WeatherBotShould
+    {
+
+        [Theory]
+        [InlineData("Nablus", 10, 11)]
+        public void Activate_RainBot_If_RainBot_Enabled_And_Humidity_GreaterThan_Threshold(string location, float temperetureThreshold, float humidityThreshold)
+        {
+            var fixture = new Fixture();
+            var rainBot = fixture.Build<RainWeatherBot>()
+                .With(x => x.Enabled, true)
+                .With(x => x.Message, "This is a rain bot")
+                .With(x => x.HumidityThreshold, 10)
+                .Create();
+
+            var data = new WeatherDTO
+            {
+                Location = location,
+                Temperature = temperetureThreshold,
+                Humidity = humidityThreshold
+            };
+
+            Assert.True(rainBot.DisplayMessage(data));
```

Comment:
don't do act and assert in the same statement.
u have to isolate and define each section by comment